### PR TITLE
Update 'Microsoft.PowerShell.CoreCLR.Eventing' to resolve conflict with 'System.Diagnostics.EventLog'

### LINKED ISSUE #10067: Updating System.ServiceProcess.ServiceController to its .Net Core 3-preview6 version breaks 2 tests
Updating the 2 NuGet package [System.ServiceProcess.ServiceController](https://www.nuget.org/packages/System.ServiceProcess.ServiceController) (or `Microsoft.Windows.Compatibility`, which is depending on it) from `4.5.0` (.Net Core 2.1) to `4.6.0-preview6.19303.8` (.Net Core 3-preview6)  breaks 2 tests and the reason for their failure seems to be unclear (i.e. whether it is a problem of PowerShell itself that is now being exposed or a regression in the `System.ServiceProcess.ServiceController`  NuGet package itself), therefore their update was deferred in PR 9888

# Steps to reproduce

Update the `System.ServiceProcess.ServiceController` NuGet package to `4.6.0-preview6.19303.8`

# Expected behavior

All tests pass.

# Actual behavior

There are 2 failures on Windows:

- Set/New/Remove-Service cmdlet tests.Set-Service can accept a ServiceController as positional input:
```
Expected strings to be the same, but they were different.
String lengths are both 14.
Strings differ at index 0.
Expected: 'newdisplayname'
But was:  'testsetservice'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1: line 321
321:             $service.DisplayName | Should -BeExactly $newdisplayname
```
It is worthwhile noting that the first Windows service test throws an `ObjectDisposedException` before it fails:
```
		Set-Service : Cannot access a disposed object.
Object name: 'ServiceController'.
At C:\Users\cberg\git\PowerShell\test\powershell\Modules\Microsoft.PowerShell.Management\Set-Service.Tests.ps1:318 char:25
+ ...       $script = { Set-Service $service -DisplayName $newdisplayname }
+                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (:) [Set-Service], ObjectDisposedException
+ FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.SetServiceCommand
```
image

- Get-WinEvent.Get-WinEvent can retrieve events.Get-WinEvent can use the simplest of filters
```
Cannot convert argument "value", with value: "System.Diagnostics.Eventing.Reader.EventLogRecord", for "TryAdd" to type "System.Diagnostics.Eventing.Reader.EventLogRecord": "Cannot convert the "System.Diagnostics.Eventing.Reader.EventLogRecord" value of type "System.Diagnostics.Eventing.Reader.EventLogRecord" to type "System.Diagnostics.Eventing.Reader.EventLogRecord"."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Diagnostics\Get-WinEvent.Tests.ps1: line 61
61:                 $testEventDict.TryAdd($te.Id, $te)
```

# Environment data

Windows

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Fix #10067

A few more namespaces are included in the latest `System.Diagnostics.EventLog` NuGet package, including `System.Diagnostics.Eventing.Reader`, which has `EventLogRecord` and more type that are related to `EventProvider`.
`Microsoft.PowerShell.CoreCLR.Eventing` contains the ported .NET code for `EventProvider` and it also contains those public types, which cause a conflict.

Update `Microsoft.PowerShell.CoreCLR.Eventing` to depend on `System.Diagnostics.EventLog` to avoid conflicts.

/cc @bergmeister 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by bergmeister** [comment]
Thanks for looking into this.
Once you get a green build, you could also try to update the WindowsCompability NuGet package (not possible at the moment because it has a dependency on the ServiceController package that needs updating first...)

File: assets/files.wxs [hunk 1]
```diff
@@ -3042,6 +3042,15 @@
       <Component Id="cmpCE991A76187F47148F1854BC02E3524E" Guid="{39a0126f-44e9-471a-83d6-139d1636e79a}">
```
Comment (PoshChan) [inline assets/files.wxs]: @daxian-dbw, your last commit had 1 failures in `PowerShell-CI-windows`
Validate Update-Help from the Web for all PowerShell modules..Validate Update-Help for module 'Microsoft.PowerShell.Core' with scope as 'False'
```powershell
Expected 1, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Help\UpdatableHelpSystem.Tests.ps1: line 181
181:     $helpFilesInstalled.Count | Should -Be $expectedHelpFiles.Count
```

**Comment by daxian-dbw** [comment]
@bergmeister The `WindowsCompability` NuGet package has been updated in this PR.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@daxian-dbw Thanks for the cleanup!

**Pull request merged**